analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -411,11 +411,17 @@ public static bool IsComment(this SyntaxTrivia trivia)
         public static ImmutableArray<ExpressionSyntax> ArgumentValuesForParameter(SemanticModel semanticModel, ArgumentListSyntax argumentList, string parameterName)
         {
             var methodParameterLookup = new CSharpMethodParameterLookup(argumentList, semanticModel);
-            if (methodParameterLookup.TryGetSyntax(parameterName, out var argumentSyntax))
-            {
-                return argumentSyntax.Select(x => x.Expression).ToImmutableArray();
-            }
-            return ImmutableArray<ExpressionSyntax>.Empty;
+            return methodParameterLookup.TryGetSyntax(parameterName, out var argumentSyntax)
+                ? argumentSyntax.Select(x => x.Expression).ToImmutableArray()
+                : ImmutableArray<ExpressionSyntax>.Empty;
         }
+
+        // (arg, b) = something
+        public static bool IsAssignmentToTuple(this ArgumentSyntax argument) =>
```

Comment:
I think the name `IsAssignmentToTuple` can be improved (did so in #3810 as well, but didn't have a better idea).

Because when looking at `(a, b) = (c, d)` we have arguments (and tuples) on both sides of the assignment, so `argument.IsAssignmentToTuple` is hard to follow. My first thought is to read as "argument is assigned to tuple" (like in `(a,b) = argument`) which is not the case. 

I suggest 
- `IsTupleArgumentOnAssignmentLeftSide` (not satisfied with it either...)
- `IsInTupleArgumentOnLeftSideOfAssignment` which would align with another name `IsInNameOfArgument()`

Reply:
`IsInTupleAssignmentTarget` as discussed offline